### PR TITLE
Check elf header for executable 

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

Verify elf header for executable for bpf_loader programs.

#### Summary of Changes

Check elf header for executable

Fixes #

### CODE REVIEW

File: sdk/src/account.rs [hunk 1]
```diff
@@ -812,8 +812,9 @@ pub fn is_executable(account: &impl ReadableAccount, feature_set: &FeatureSet) -
     if !feature_set.is_active(&deprecate_executable_meta_update_in_bpf_loader::id()) {
         account.executable()
     } else {
-        // First, check if the account is empty. Empty accounts are not executable.
-        if account.data().is_empty() {
+        // First, check if the account is empty or all zeros. Empty or all zeros
+        // accounts are not executable.
+        if account.data().is_empty() || account.data().iter().all(|&x| x == 0) {
```
Comment (Lichtso) [inline sdk/src/account.rs:817]: That check is too expensive, better to test if the ELF header exists in RBPF.

Comment (HaoranYi) [reply sdk/src/account.rs:817]: yeah. That's sound like a better idea.

**Comment by willhickey** [comment]
This repository is no longer in use. Please re-open this pull request in the agave repo: https://github.com/anza-xyz/agave

**Comment by HaoranYi** [comment]
Moved to [anza/agave](https://github.com/anza-xyz/agave/pull/89)

**Pull request closed without merging**